scripts/feature_tests/upgrade/upgrade_vars.sh

```diff
@@ -23,3 +23,7 @@ export CAPM3_API_VERSION="v1alpha4"
 
 export KUBERNETES_VERSION="v1.21.1"
 export UPGRADED_K8S_VERSION="v1.21.2"
+
+export NUM_NODES="4"
```

Comment:
One problem with putting `NUM_NODES` here is that it will prevent overriding this value from the Jenkins jobs:

https://github.com/metal3-io/project-infra/blob/60aafbe4bcf82f2233c6c7bd3be1335dbd5deadd/jenkins/scripts/integration_test.sh#L37

(I'm not sure this is such a useful feature, especially since it seems the replica numbers are hardcoded elsewhere, but it is there.)

Reply:
That is a good point. My reasoning was that since we already hardcode `NUM_OF_MASTER_REPLICAS` and `NUM_OF_WORKER_REPLICAS` we might as well do the same with `NUM_NODES`. But maybe it would be better to make them all configurable? Not quite sure where or how though. :thinking:

Reply:
I'm all in favour of having things be less configurable and 'just work'.

Usually I would set these variables in `<username>_config.sh`, I didn't realise before that they were hardcoded in the case of upgrade tests.

I am sure there is a sophisticated way to do this, but 'fail fast' is a good pattern in Ansible that means you don't have to guess the user's intent about what they want to do in their environment.

Reply:
Agreed! Updated now to fail fast instead of hardcoding `NUM_NODES`